src/strategies/build_strategy.rs

```diff
@@ -0,0 +1,413 @@
+                    .arg("build")
+                    .arg("-t")
+                    .arg(image)
+                    .status()?;
+
+                if status.success() {
+                    info!("Successfully built {image}");
+                } else {
+                    bail!("Failed to build {image}");
+                }
+            }
+            #[cfg(feature = "builtin-podman")]
+            Strategy::Socket(client, rt) => {
+                rt.block_on(async {
+                    let signals = Signals::new([SIGTERM, SIGINT, SIGQUIT])?;
+                    let handle = signals.handle();
+
+                    let (kill_tx, mut kill_rx) = oneshot::channel::<()>();
+
+                    let signals_task =
+                        tokio::spawn(handle_signals(signals, kill_tx, self.uuid, client.clone()));
+
+                    // Get podman ready to build
+                    let opts = ImageBuildOpts::builder(".")
+                        .tag(image)
+                        .dockerfile("Containerfile")
+                        .remove(true)
+                        .layers(true)
+                        .labels([(BUILD_ID_LABEL, self.uuid.to_string())])
+                        .pull(true)
+                        .build();
+                    trace!("Build options: {opts:#?}");
+
+                    info!("Building image {image}");
+                    match client.images().build(&opts) {
+                        Ok(mut build_stream) => loop {
+                            tokio::select! {
+                                Some(chunk) = build_stream.next() => {
+                                    match chunk {
+                                        Ok(chunk) => chunk
+                                            .stream
+                                            .trim()
+                                            .lines()
+                                            .map(str::trim)
+                                            .filter(|line| !line.is_empty())
+                                            .for_each(|line| info!("{line}")),
+                                        Err(e) => bail!("{e}"),
+                                    }
+                                },
+                                _ = &mut kill_rx => {
+                                    break;
+                                },
+                                else => {
+                                    break;
+                                }
+                            }
+                        },
+                        Err(e) => bail!("{e}"),
+                    };
+                    handle.close();
+                    signals_task.await?;
+                    Ok(())
+                })?;
+            }
+        };
+        Ok(())
+    }
+
+    pub fn tag(&self, src_image: &str, image_name: &str, tag: &str) -> Result<()> {
+        let dest_image = format!("{image_name}:{tag}");
+
+        match &self.strategy {
```

Comment:
So I notice we are matching alot. What if we create a trait like our clap commands and have them specify generic commands we can call?